### PR TITLE
Fix Resources issues with builder.

### PR DESCRIPTION
**Pull request by ryanwilson**
- Removes framework-specific code for resources
- Embeds all resources into .framework folders first to prep for
Carthage
- Extracts all resources into existing zip structure
- Fixes missing resources from cached OSS framework

### CODE REVIEW

**Comment by ryanwilson** [comment]
This should fix internal bug b/128853130 as well.

File: ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 1]
```diff
@@ -75,12 +75,11 @@ struct FrameworkBuilder {
...
   ///               be used.
   ///   - cacheEnabled: Flag for enabling the cache. Defaults to false.
-  /// - Returns: A URL to the framework that was built (or pulled from the cache) and a URL to the
-  ///     Resources directory containing all required bundles.
+  /// - Returns: A URL to the framework that was built (or pulled from the cache).
   public func buildFramework(withName podName: String,
                              version: String,
                              cacheKey: String?,
-                             cacheEnabled: Bool = false) -> (framework: URL, resources: URL) {
+                             cacheEnabled: Bool = false) -> URL {
```
Comment (paulb777) [inline ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift]: Nice API improvement!

File: ZipBuilder/Sources/ZipBuilder/ResourcesManager.swift [hunk 2]
```diff
@@ -80,28 +161,22 @@ extension ResourcesManager {
   ///                  directory if it doesn't exist.
   /// - Returns: An array of URLs pointing to the newly located bundles.
   /// - Throws: Any file system errors that occur.
+  @discardableResult
```
Comment (paulb777) [inline ZipBuilder/Sources/ZipBuilder/ResourcesManager.swift]: I learned something!

Comment (ryanwilson) [reply ZipBuilder/Sources/ZipBuilder/ResourcesManager.swift]: :D it's very nice, especially for testing (which would be nice to add eventually too...) - use the result in the testing but if you don't need it at the call site it's handy 🎉

File: ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift (hunk 1 above)
Comment (ryanwilson) [reply ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift]: Thanks, I agree! 😄

**Pull request merged**